langchain4j-elastic/src/main/java/dev/langchain4j/store/embedding/elastic/Document.java

```diff
@@ -0,0 +1,26 @@
+package dev.langchain4j.store.embedding.elastic;
+
+import dev.langchain4j.data.document.Metadata;
+import lombok.Builder;
+import lombok.Data;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Elasticsearch document object, for the purpose of construct document object from embedding and text segment
+ */
+@Data
+@Builder
+public class Document {
+
+    private List<Float> vector;
+    private String text;
+    private Metadata metadata;
```

Comment:
should it be Map<String,String> for more simplicity?

Reply:
OK. That's what I originally did, but `Metadata` do not expose method to get `Map<String, String> metadata` so I replace `Map<String, String>` to `Metadata`. Should it possible to add method `public Map<String, String> getOriginalMetadata()` in `Metadata`?